app/src/main/java/org/wikipedia/pageimages/PageImageDatabaseTable.kt

```diff
@@ -0,0 +1,63 @@
+package org.wikipedia.pageimages
+
+import android.content.ContentValues
+import android.database.Cursor
+import org.wikipedia.database.DatabaseTable
+import org.wikipedia.database.column.Column
+import org.wikipedia.database.contract.PageImageHistoryContract
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.page.PageTitle
+
+// todo: network caching preserves images. Remove this class and drop table?
+class PageImageDatabaseTable : DatabaseTable<PageImage>(PageImageHistoryContract.TABLE, PageImageHistoryContract.Image.URI) {
+    override fun fromCursor(cursor: Cursor): PageImage {
+        val wiki = WikiSite(PageImageHistoryContract.Col.SITE.`val`(cursor), PageImageHistoryContract.Col.LANG.`val`(cursor))
+        val title = PageTitle(PageImageHistoryContract.Col.NAMESPACE.`val`(cursor), PageImageHistoryContract.Col.API_TITLE.`val`(cursor), wiki)
+        val imageName = PageImageHistoryContract.Col.IMAGE_NAME.`val`(cursor)
+        title.setDisplayText(PageImageHistoryContract.Col.DISPLAY_TITLE.`val`(cursor))
```

Comment:
Looks like the `val()` is a reserved word that requires special quotes around it. Would it be a great opportunity to also convert `StrColumn.java` to Kotlin and also rename the `val()` method?

Reply:
I addressed this in a separate PR #2216  Just so it is easy to review since that name change turned out to be a big chunk of change.. 😅